perf/perf_hv_gpci.py

```diff
@@ -80,20 +93,39 @@ def run_cmd(self, cmd):
         if output.exit_status != 0:
             self.fail_cmd.append(cmd)
 
-    def test_gpci_events(self):
+    def gpci_events(self, val):
         perf_stat = "perf stat"
-        perf_flags = '-C 1 -v -e'
-
-        for line in self.list_of_hv_gpci_events:
-            evt = "hv_gpci/%s,hw_chip_id=12/" % line
-            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, evt)
+        perf_flags = '-v -e'
+        for line in val:
+            if line in self.list_phys:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'phys_processor_idx=?', 'phys_processor_idx=1'))
+            if line in self.list_sibling:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'sibling_part_id=?', 'sibling_part_id=2'))
+            if line in self.list_partition:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'partition_id=?', 'partition_id=1'))
+            if line in self.list_hw:
+                line = "%s,%s/" % (line.split(',')[0], line.split(',')[1].replace(
+                    'hw_chip_id=?', 'hw_chip_id=12'))
+            if line in self.list_noid:
+                line = "%s/" % line
+
+            cmd = "%s %s %s sleep 1" % (perf_stat, perf_flags, line)
             self.run_cmd(cmd)
-            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "%s --per-core -a -e %s sleep 1" % (perf_stat, line)
             self.run_cmd(cmd)
-            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, evt)
+            cmd = "%s --per-socket -a -e %s sleep 1" % (perf_stat, line)
             self.run_cmd(cmd)
+            self.error_check()
 
-        self.error_check()
+    def test_gpci_events(self):
+        self.gpci_events(self.list_phys)
+        self.gpci_events(self.list_sibling)
+        self.gpci_events(self.list_partition)
+        self.gpci_events(self.list_hw)
+        self.gpci_events(self.list_noid)
```

Comment:
@disgoel why cannot we use a loop here?

Reply:
with loop its taking old events list and giving parser error

Reply:
@disgoel I mean in following fashion

```
events = ['self.list_phys',  'self.list_sibling', 'self.list_partition', 'self.list_hw', 'self.list_noid']
for even in events:
    self.gpci_events(even)
```